code/datums/components/growth_and_differentiation.dm

```diff
@@ -44,6 +52,10 @@
 	src.optional_checks = optional_checks
 	src.optional_grow_behavior = optional_grow_behavior
 
+	if(islist(signals_to_kill_on))
+		src.signals_to_kill_on = signals_to_kill_on
+		RegisterSignals(parent, src.signals_to_kill_on.Copy(), CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(qdel), src))
```

Comment:
uh, what?

Reply:
what's wrong specifically

Reply:
oh it doesn't work at all one second

Reply:
for some reason i thought i was doing timer stuff